app/frontend/src/cljs/teet/admin/indexes_view.cljs

```diff
@@ -0,0 +1,343 @@
+  ""
+  [e! page-state index edit? edit-values?]
+    (let [values (:cost-index/values index)
+          index-valid (t/date-time (:cost-index/valid-from index))]
+     [:div
+      [:div {:class [(<class common-styles/margin-bottom 2)
+                     (<class common-styles/flex-align-center)]}
+       [:div {:style {:min-width "600px"}}
+        (if edit?
+          [edit-index-form e! index (:edit-index page-state)]
+          [typography/Heading1
+            (str (:cost-index/name index))])]
+
+       (when (not (or edit? edit-values?))
+         [buttons/button-secondary {:id (str "admin-editindex")
+                                    :on-click (e! admin-controller/->EditIndexForm)}
+          (tr [:buttons :edit-index])])]
+      [:div {:class (<class common-styles/flex-row)}
+       [:div {:style {:min-width "250px"}}
+        [:b (tr [:fields :cost-index/type])]]
+       (tr-enum (:cost-index/type index))]
+      [:div {:class [(<class common-styles/flex-row)
+                     (<class common-styles/margin-bottom 2)]}
+       [:div {:style {:min-width "250px"}}
+        [:b (tr [:fields :cost-index/valid-from])]] (str
+                                                             (tr [:calendar :months (dec (t/month index-valid))])
+                                                             " "
+                                                             (t/year index-valid))]
+  (if edit-values?
+    [:div [edit-index-values e! (:edit-index-values page-state) index values]]
+    [:div
+     (if (seq values)
+       [view-index-values e! index values]
+       [:div {:class [(<class common-styles/flex-row)
+                      (<class common-styles/margin-bottom 2)]} (tr [:indexes-admin :no-values-entered])])
+     (when-not edit?
+       [buttons/button-primary {:id (str "admin-editindexvalues")
+                                :on-click (e! admin-controller/->EditIndexValues)}
+      (tr [:buttons :edit-index-values])])])]))
+
+(defn indexes-content
+  [e! query page-state id]
+  (let [add-form (or (:add-index page-state) (:add-index-form query))
+        index-data (:index-data page-state)]
+    [:<>
+     (when add-form
+      [add-index-form e! {:form-value add-form
+                          :on-change-event admin-controller/->UpdateAddIndexForm
+                          :save-event admin-controller/->SaveIndex
+                          :cancel-event admin-controller/->CancelIndex}])
+    (when id
+      [view-index-info e! page-state (first
+                            (filterv
+                              #(when (= id (str (:db/id %))) %)
+                              index-data))
+       (some? (:edit-index page-state))
+       (some? (:edit-index-values page-state))])]))
+
+(defn indexes-page
+  "The main indexes admin page"
+  [e! {params :params
+       route :route
+       query :query} page-state]
+  (let [id (:id params)]
+   [:div {:class (<class common-styles/flex-align-end)}
+   [query/debounce-query
```

Comment:
why is there another query here? couldn't the same query that returns the page-state return this information as well?

Reply:
Leftover from initial implementation. Removed.